core/src/main/java/org/hibernate/ogm/datastore/document/association/spi/AssociationRows.java

```diff
@@ -28,7 +27,7 @@
 	private final Map<RowKey, AssociationRow<?>> rows;
 
 	public AssociationRows(AssociationKey associationKey, Collection<?> wrapped, AssociationRowFactory associationRowFactory) {
-		this.rows = newHashMap( wrapped.size() );
+		this.rows = new LinkedHashMap<>( wrapped.size() );
```

Comment:
Perhaps, but this is preexisting code, why stopping this PR that tries to solve a different issue?